src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,95 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final int VALUE_INDEX = 1;
+    private static final int WORD_INDEX = 0;
+    private static final int SUPPLY_INDEX = 0;
+    private static final int BUY_INDEX = 1;
+    private static final int RESULT_INDEX = 2;
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT_VALUE = "result";
+    private static final String ZERO_VAR = "0";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String statistic = readText(fromFileName);
+        writeText(toFileName, statistic);
+    }
+
+    private void writeText(String toFileName, String statistic) {
+        try {
+            Files.writeString(new File(toFileName).toPath(), statistic);
+        } catch (IOException e) {
+            throw new RuntimeException("can't write to file: " + toFileName, e);
+        }
+    }
+
+    private String readText(String fromFileName) {
+        List<String> list;
+        try {
+            list = Files.readAllLines(new File(fromFileName).toPath());
+        } catch (IOException e) {
+            throw new RuntimeException("wrong file path: " + fromFileName, e);
+        }
+        return getResultText(list);
+    }
+
+    private String getResultText(List<String> list) {
```

Comment:
here let's just use for-each loop
introduce variables totalSupply and totalBuy

for each line, split line by comma, and according to the operation - add value either to totalSupply, or to totalBuy

then use StringBuilder to create a ready String report